src/jit.c

```diff
@@ -236,12 +238,27 @@ static int temp_reg = R8;
  *   r24       Temp - used for generating 32-bit immediates
  *   r25       Temp - used for modulous calculations
  */
-static const int register_map[] = {
-    R5, R6, R7, R9, R11, R12, R13, R14, R15, R16, R17, R18, R26,
+static struct host_reg register_map[] = {
+    {R5, 0},  {R6, 0},  {R7, 0},  {R9, 0},  {R11, 0}, {R12, 0}, {R13, 0},
+    {R14, 0}, {R15, 0}, {R16, 0}, {R17, 0}, {R18, 0}, {R26, 0},
 };
-static inline void emit_load_imm(struct jit_state *state, int dst, int64_t imm);
 #endif
 
+static const int n_reg =
```

Comment:
Use `n_regs` for consistency. However, it is a bit confusing for the use of host and virtual registers. Can you clarify?